### PR TITLE
Add TLS protocol support

### PR DESCRIPTION
**Pull request by zaharidichev**
This change adds support for TLS in the `ProxyProtocol` enum. For the time being we can only use SNIs
to do do TLS routing.

Signed-off-by: Zahari Dichev <zaharidichev@gmail.com>

### CODE REVIEW

**Comment by cratelyn** [comment]
```
    = Solution: Upgrade to >=0.12.3 (try `cargo update -p tonic`)
   = tonic v0.10.2
     └── linkerd2-proxy-api v0.14.0
```

this advisory is only two days old, so i suspect it's unrelated to your changes here.

**Comment by cratelyn** [comment]
`note: this only affects v0.12.0 - v0.12.2` from the advisory, here: https://github.com/hyperium/tonic/security/advisories/GHSA-4jwc-w2hc-78qv. we're unaffected, via use of `v0.10.2`.

File: proto/outbound.proto [hunk 1]
```diff
@@ -308,6 +316,45 @@ message OpaqueRoute {
...
+
+  // Must have at least one rule.
+  repeated Rule rules = 3;
+  message Rule {
+    Distribution backends = 1;
+  }
+
+  message Distribution {
+    oneof kind {
+      Empty empty = 1;
```
Comment (sfleen) [inline proto/outbound.proto:333]: Is an empty variant strictly necessary here? We should be able to use presence of the other variants to indicate empty.

Comment (cratelyn) [reply proto/outbound.proto:333]: https://github.com/linkerd/linkerd2-proxy-api/blob/main/proto/outbound.proto#L139 i was curious myself, so here's an existing example of that pattern, just to demonstrate that pattern

Comment (olix0r) [reply proto/outbound.proto:333]: This is so that this state is explicit and can't be ambiguous i.e. if a field is unset.

File: Cargo.toml [hunk 2]
```diff
@@ -16,6 +16,7 @@ default = []
 arbitrary = ["quickcheck"]
 destination = ["http-types", "net", "prost-types", "tonic/codegen"]
 grpc-route = ["http-route"]
+tls-route = []
```
Comment (olix0r) [inline Cargo.toml:19]: nit: sort

**Review by olix0r — APPROVED** [review]

**Pull request merged**